Surface error results to BuildRun (#930)
shipwright-io/build: docs/buildstrategies.md

 # [...]
  output:
    digest: sha256:07626e3c7fdd28d5328a8d6df8d29cd3da760c7f5e2070b534f9b880ed093a53
    size: 1989004
  # [...]
```

> Additionally, you can store error details for debugging purposes in case that a BuildRun fails using your strategy.

Reviewer: ```suggestion
Additionally, you can store error details for debugging purposes when a BuildRun fails using your strategy.
```
